rasa/telemetry.py

```diff
@@ -0,0 +1,506 @@
+
+    if not telemetry_id:
+        logger.debug("Will not report telemetry events as no ID was found.")
+        return
+
+    if not properties:
+        properties = {}
+
+    properties[TELEMETRY_ID] = telemetry_id
+
+    try:
+        await _send_event(
+            telemetry_id, event_name, properties, with_default_context_fields(context)
+        )
+    except Exception as e:  # skipcq:PYL-W0703
+        logger.debug(
+            f"An error occurred when trying to report the telemetry event: {e}"
+        )
+
+
+def get_telemetry_id() -> Optional[Text]:
+    """Return the unique telemetry identifier for this Rasa Open Source install.
+    The identifier can be any string, but it should be a UUID.
+
+    Returns:
+        The identifier, if it is configured correctly.
+    """
+
+    try:
+        telemetry_config = (
+            rasa_utils.read_global_config_value(CONFIG_FILE_TELEMETRY_KEY) or {}
+        )
+
+        return telemetry_config.get(CONFIG_TELEMETRY_ID)
+    except Exception as e:  # skipcq:PYL-W0703
+        logger.debug(f"Unable to retrieve telemetry ID: {e}")
+        return None
+
+
+def toggle_telemetry_reporting(is_enabled: bool) -> None:
+    """Write to the configuration if telemetry tracking should be enabled or disabled.
+
+    Args:
+        is_enabled: `True` if the telemetry reporting should be enabled,
+            `False` otherwise.
+    """
+
+    configuration = rasa_utils.read_global_config_value(CONFIG_FILE_TELEMETRY_KEY)
+
+    if configuration:
+        configuration[CONFIG_TELEMETRY_ENABLED] = is_enabled
+    else:
+        configuration = _default_telemetry_configuration(is_enabled)
+
+    rasa_utils.write_global_config_value(CONFIG_FILE_TELEMETRY_KEY, configuration)
+
+
+@ensure_telemetry_enabled
+async def track_model_training(
+    training_data: TrainingDataImporter, model_type: Text
+) -> None:
+    """Tracks when a new Git repository was connected to Rasa X server.
+
+    Args:
+        training_data: Training data used for the training.
+        model_type: Specifies the type of training, should be either "rasa", "core"
+            or "nlu".
+    """
+
+    config = await training_data.get_config()
+    stories = await training_data.get_stories()
+    nlu_data = await training_data.get_nlu_data()
+    domain = await training_data.get_domain()
+
+    await track(
+        TRAINING_STARTED_EVENT,
+        {
+            "language": config.get("language"),
+            "model_type": model_type,
+            # TODO: figure out how to track these
+            # "pipeline": config.get("pipeline"),
+            # "policies": config.get("policies"),
+            "num_intent_examples": len(nlu_data.intent_examples),
+            "num_entity_examples": len(nlu_data.entity_examples),
+            "num_actions": len(domain.action_names),
+            # Old nomenclature from when 'responses' were still called
+            # 'templates' in the domain
+            "num_templates": len(domain.templates),
```

Comment:
Can't we use the new name here? Given that telemetry is new for OSS.

Reply:
I wanted to keep the two events the same to have consistent naming in our telemetry db. I mean otherwise we are going to have different names in the db schema which isn't great either.